src/search/landmarks/landmark_cost_partitioning_heuristic.cc

```diff
@@ -0,0 +1,159 @@
+
+    double h_val = lm_cost_assignment->cost_sharing_h_value(*lm_status_manager);
+    if (h_val == numeric_limits<double>::max()) {
+        return DEAD_END;
+    } else {
+        return static_cast<int>(ceil(h_val - epsilon));
+    }
+}
+
+bool LandmarkCostPartitioningHeuristic::dead_ends_are_reliable() const {
+    return true;
+}
+
+class LandmarkCostPartitioningHeuristicFeature : public plugins::TypedFeature<Evaluator, LandmarkCostPartitioningHeuristic> {
+public:
+    LandmarkCostPartitioningHeuristicFeature() : TypedFeature("lmcp") {
+        document_title("Landmark cost partitioning heuristic");
+        document_synopsis(
+            "See the papers" +
+            utils::format_conference_reference(
+                {"Erez Karpas", "Carmel Domshlak"},
+                "Cost-Optimal Planning with Landmarks",
+                "https://www.ijcai.org/Proceedings/09/Papers/288.pdf",
+                "Proceedings of the 21st International Joint Conference on "
+                "Artificial Intelligence (IJCAI 2009)",
+                "1728-1733",
+                "AAAI Press",
+                "2009") +
+            "and" +
+            utils::format_conference_reference(
+                {"Emil Keyder and Silvia Richter and Malte Helmert"},
+                "Sound and Complete Landmarks for And/Or Graphs",
+                "https://ai.dmi.unibas.ch/papers/keyder-et-al-ecai2010.pdf",
+                "Proceedings of the 19th European Conference on Artificial "
+                "Intelligence (ECAI 2010)",
+                "335-340",
+                "IOS Press",
+                "2010"));
+
+        LandmarkHeuristic::add_options_to_feature(*this);
+        add_option<bool>(
+            "optimal",
+            "use optimal (LP-based) cost sharing",
+            "false");
+        add_option<bool>("alm", "use action landmarks", "true");
+        lp::add_lp_solver_option_to_feature(*this);
+
+        document_note(
+            "Optimal search",
+            "You probably also want to add this heuristic as a lazy_evaluator "
+            "in the A* algorithm to improve heuristic estimates.");
+        document_note(
+            "Note",
+            "To use ``optimal=true``, you must build the planner with LP "
+            "support. See LPBuildInstructions.");
+
+        document_language_support(
+            "action costs",
+            "supported");
+        document_language_support(
+            "conditional_effects",
+            "not supported");
+        document_language_support(
+            "axioms",
+            "not allowed");
+
+        document_property(
+            "admissible",
+            "yes");
+        /* TODO: This was "yes with admissible=true and optimal cost
+            partitioning; otherwise no" before. Can we answer this now that
+            the heuristic only cares about admissible? */
+        document_property(
+            "consistent",
+            "complicated; needs further thought");
```

Comment:
I discussed this with Malte. Any cost partitioning is consistent. However, the definition of consistent does not really make sense for path-dependent heuristics. For example if we expand a state s with h=5 then all children will be consistent with this heuristic estimate since we incorporate the landmark information from s in its successors. But we might later find a different path to s that increases its heuristic value to let's say 10. Then the heuristic can become inconsistent as long as we don't reevaluate the heuristic values of the successors (and this can lead to a chain reaction). It also depends on whether or not you use lazy_evaluator.

Suggestion (not discussed with Malte): We write "(yes), see document note about consistency" here and add a document_note, maybe something like this:
Consistency
The heuristic is consistent along single paths if it is set as lazy_evaluator; i.e. when expanding s then we have h(s) <= h(s')+cost(a) for all successors s' of s reached with a. But newly found paths to s can increase h(s), at which point the above inequality might not hold anymore."

Reply:
This is also how I explained it to myself. 

I like the suggestion to add a document note, but I think "no, see document note about consistency" would be more accurate. My argument also relies on the assumption that when someone looks into the documentation to check whether a heuristic is consistent, they hope to find "consistent=yes" rather than "no". If they find "(yes), see document note ..." it is more tempting to skip the document note than when we say no.

Reply:
Yeah that's probably better. I thought to add yes to convey that in some sense it is consistent, but the information that in some sense it's not is more imporant I agree.